src/adapters/brightcom.js

```diff
@@ -0,0 +1,214 @@
+        var siteDomain = window.location.host;
+        var sitePage = window.location.pathname + location.search + location.hash;
+
+        // Prepare impressions object
+        var brightcomImps = [];
+
+        // Prepare a variable for publisher id
+        var pubId = '';
+
+        // Go through the requests and build array of impressions
+        utils._each(bidRequests, function(bid) {
+
+            // Get impression details
+            var tagId = utils.getBidIdParamater('tagId', bid.params);
+            var ref = utils.getBidIdParamater('ref', bid.params);
+            var adWidth=0;
+            var adHeight=0;
+
+            // If no publisher id is set, use the current
+            if (pubId === '') {
+                // Get the current publisher id (if it doesn't exist, it'll return '')
+                pubId = utils.getBidIdParamater('pubId', bid.params);
+            }
+
+            // Brightcom supports only 1 size per impression
+            // Check if the array contains 1 size or array of sizes
+            if (bid.sizes.length === 2 && typeof bid.sizes[0] === 'number' && typeof bid.sizes[1] === 'number') {
+                // The array contains 1 size (the items are the values)
+                adWidth = bid.sizes[0];
+                adHeight = bid.sizes[1];
+            } else {
+                // The array contains array of sizes, use the first size
+                adWidth = bid.sizes[0][0];
+                adHeight = bid.sizes[0][1];
+            }
+
+            // Build the impression
+            var imp = {
+                id: utils.getUniqueIdentifierStr(),
+                banner: {
+                    w: adWidth,
+                    h: adHeight
+                },
+                tagid: tagId
+            };
+
+            // If ref exists, create it (in the "ext" object)
+            if (ref !== '') {
+                imp.ext = {
+                    refoverride: ref
+                };
+            }
+
+            // Add current impression to collection
+            brightcomImps.push(imp);
+            // Add mapping to current bid via impression id
+            bidmanager.pbCallbackMap[imp.id] = bid;
+
+            // Add current ad unit's code to tracking
+            reqAdUnitsCode.push(bid.placementCode);
+
+        });
+
+        // Build the bid request
+        var brightcomBidReq = {
+            id: utils.getUniqueIdentifierStr(),
+            imp: brightcomImps,
+            site:{
+                publisher: {
+                    id: pubId
+                },
+                domain: siteDomain,
+                page: sitePage
+            }
+        };
+
+        // Add timeout data, if available
+        var PREBID_TIMEOUT = PREBID_TIMEOUT || 0;
+        var curTimeout = PREBID_TIMEOUT;
+        if (curTimeout > 0) {
+            brightcomBidReq.tmax = curTimeout;
+        }
+
+        // Define the bid request call URL
+        var bidRequestCallUrl = '//' + brightcomUrl +
```

Comment:
We are standardizing HTTPS for all adapters so please update any relative protocol code to always use HTTPS.